torchio/transforms/preprocessing/spatial/resample.py

```diff
@@ -94,18 +112,46 @@ def parse_interpolation(interpolation: Interpolation) -> int:
         return order
 
     @staticmethod
-    def check_reference_image(reference_image: str, sample: Subject):
-        if not isinstance(reference_image, str):
-            message = f'reference_image argument should be of type str, type {type(reference_image)} was given'
+    def check_affine(affine_name: str, image_dict: dict):
+        if not isinstance(affine_name, str):
+            message = (
+                'Affine name argument must be a string,'
+                f' not {type(affine_name)}'
+            )
             raise TypeError(message)
-        if reference_image not in sample.keys():
-            message = f'reference_image=\'{reference_image}\' not present in sample, only these keys were found: {sample.keys()}'
-            raise ValueError(message)
+        if affine_name in image_dict:
+            matrix = image_dict[affine_name]
+            if not isinstance(matrix, np.ndarray):
+                message = (
+                    'The affine matrix must be a NumPy array,'
+                    f' not {type(matrix)}'
+                )
+                raise TypeError(message)
+            if matrix.shape != (4, 4):
+                message = (
+                    'The affine matrix shape must be (4, 4),'
+                    f' not {matrix.shape}'
+                )
+                raise ValueError(message)
+
+    @staticmethod
+    def check_affine_key_presence(affine_name: str, sample: Subject):
+        for image_dict in sample.get_images(intensity_only=False):
+            if affine_name in image_dict:
+                return
+        message = (
+            f'An affine name was given ("{affine_name}"), but it was not found'
+            ' in any image in the sample'
+        )
+        raise ValueError(message)
 
     def apply_transform(self, sample: Subject) -> dict:
         use_reference = self.reference_image is not None
-        iterable = sample.get_images_dict(intensity_only=False).items()
-        for image_name, image_dict in iterable:
+        use_pre_affine = self.affine_name is not None
+        sample_images = sample.get_images_dict(intensity_only=False)
+        if use_pre_affine:
+            self.check_affine_key_presence(self.affine_name, sample)
+        for image_name, image_dict in sample_images.items():
```

Comment:
That's a tiny detail but maybe this would be a bit clearer:
```python
use_pre_affine = self.affine_name is not None
if use_pre_affine:
    self.check_affine_key_presence(self.affine_name, sample)
iterable = sample.get_images_dict(intensity_only=False).items()
for image_name, image_dict in iterable:
```